src/controller/CHIPDeviceController.h

```diff
@@ -327,6 +327,8 @@ class DLL_EXPORT DeviceController : public SessionRecoveryDelegate
      */
     NodeId GetNodeId() const { return mLocalId.GetNodeId(); }
 
+    FabricInfo * GetFabricInfo() { return mFabricInfo; }
```

Comment:
This is actually completely legitimate, and should always be available since otherwise it would mean a controller doesn't have the whole set of credentials needed for operational communication.

Reply:
I don't like it either, but I haven't found a better alternative.